### PR TITLE
Added move/copy support to API model

### PR DESCRIPTION
**Pull request by kasvith**
Pull Request for Issue # .

### Summary of Changes
- Added move/copy functions to API

### CODE REVIEW

File: administrator/components/com_media/models/api.php [hunk 1]
```diff
@@ -214,4 +214,40 @@ private function getSafeName($name)
 
 		return $name;
 	}
+
+	/**
+	 * Copies file of folder from source path to destination path
+	 * If forced, exciting files/folders would be overwritten
```
Comment (dneukirchen) [inline administrator/components/com_media/models/api.php]: Typo "existing"?

File: administrator/components/com_media/models/api.php [hunk 2]
```diff
@@ -214,4 +214,40 @@ private function getSafeName($name)
...
+	 * @throws  Exception
+	 */
+	public function copy($sourcePath, $destinationPath, $force = false)
+	{
+		$this->adapter->copy($sourcePath, $destinationPath, $force);
+	}
+
+	/**
+	 * Moves file of folder from source path to destination path
+	 * If forced, exciting files/folders would be overwritten
```
Comment (dneukirchen) [inline administrator/components/com_media/models/api.php]: Typo "existing"?

File: administrator/components/com_media/models/api.php [hunk 3]
```diff
@@ -214,4 +214,40 @@ private function getSafeName($name)
 
 		return $name;
 	}
+
+	/**
+	 * Copies file of folder from source path to destination path
+	 * If forced, exciting files/folders would be overwritten
+	 *
+	 * @param   string  $sourcePath       Source path of the file or folder(relative)
```
Comment (dneukirchen) [inline administrator/components/com_media/models/api.php]: missing whitespace before (relative)

File: administrator/components/com_media/models/api.php [hunk 4]
```diff
@@ -214,4 +214,40 @@ private function getSafeName($name)
...
+	 * @param   string  $destinationPath  Destination path(relative)
```
Comment (dneukirchen) [inline administrator/components/com_media/models/api.php]: missing whitespace

File: administrator/components/com_media/models/api.php [hunk 5]
```diff
@@ -214,4 +214,40 @@ private function getSafeName($name)
...
+	{
+		$this->adapter->copy($sourcePath, $destinationPath, $force);
+	}
+
+	/**
+	 * Moves file of folder from source path to destination path
+	 * If forced, exciting files/folders would be overwritten
+	 *
+	 * @param   string  $sourcePath       Source path of the file or folder(relative)
+	 * @param   string  $destinationPath  Destination path(relative)
```
Comment (dneukirchen) [inline administrator/components/com_media/models/api.php]: missing whitespace

File: administrator/components/com_media/models/api.php [hunk 6]
```diff
@@ -214,4 +214,40 @@ private function getSafeName($name)
...
+	public function copy($sourcePath, $destinationPath, $force = false)
+	{
+		$this->adapter->copy($sourcePath, $destinationPath, $force);
+	}
+
+	/**
+	 * Moves file of folder from source path to destination path
+	 * If forced, exciting files/folders would be overwritten
+	 *
+	 * @param   string  $sourcePath       Source path of the file or folder(relative)
```
Comment (dneukirchen) [inline administrator/components/com_media/models/api.php]: missing whitespace

**Review by dneukirchen — COMMENTED** [review]
Just some typo and code style issues, rest looks fine. thank you kasun

File: administrator/components/com_media/models/api.php (hunk 6 above)
Comment (kasvith) [reply administrator/components/com_media/models/api.php]: at where :/

Comment (dneukirchen) [reply administrator/components/com_media/models/api.php]: before (relative)...

**Comment by laoneo** [comment]
Thanks @kasvith!

**Comment by dneukirchen** [comment]
Good job.

**Pull request merged**